packages/swap/src/routes/v2/quote.ts

```diff
@@ -89,77 +83,36 @@ const adjustDcaQuote = ({
   };
 
   dcaQuote.depositAmount = originalDepositAmount.toString();
+  if (dcaQuote.intermediateAmount) {
+    dcaQuote.intermediateAmount = new BigNumber(dcaQuote.intermediateAmount)
```

Comment:
this is the fix - the other changes are a small refactoring to not implement this fix for the normal quote and the boos quote separately